openedx/core/djangoapps/programs/models.py

```
@@ -29,31 +43,66 @@ class ProgramsApiConfig(ConfigurationModel):
+        Generate a URL based on public service URL and API version number.
+        """
+        return urljoin(self.public_service_url, '/api/v{}/'.format(self.api_version_number))
+
+    @property
+    def authoring_app_config(self):
+        """
+        Returns a named tuple containing information required for working with the Programs
+        authoring app, a Backbone app hosted by the Programs service.
         """
-        return urljoin(self.public_service_url, "/api/v{}/".format(self.api_version_number))
+        js_url = urljoin(self.public_service_url, self.authoring_app_js_path)
+        css_url = urljoin(self.public_service_url, self.authoring_app_css_path)
+
+        return AuthoringAppConfig(js_url=js_url, css_url=css_url)
+
+    @property
+    def is_cache_enabled(self):
+        """Whether responses from the Programs API will be cached."""
+        return self.cache_ttl > 0
+
+    def get_cache_key(self, user):
+        """Construct the appropriate cache key for the given user."""
+        return self.CACHE_KEY_ROOT + '.staff' if user.is_staff else self.CACHE_KEY_ROOT + '.student'
 
     @property
     def is_student_dashboard_enabled(self):
         """
-        Indicate whether LMS dashboard functionality related to Programs should
+        Indicates whether LMS dashboard functionality related to Programs should
         be enabled or not.
         """
         return self.enabled and self.enable_student_dashboard
 
     @property
-    def is_cache_enabled(self):
-        """Whether responses from the Programs API will be cached."""
-        return self.enabled and self.cache_ttl > 0
+    def is_studio_tab_enabled(self):
+        """
+        Indicates whether Studio functionality related to Programs should
+        be enabled or not.
+        """
+        return (
+            self.enabled and
+            self.enable_studio_tab and
+            bool(self.authoring_app_js_path) and
+            bool(self.authoring_app_css_path)
```

that makes sense but I fear we are conflating explicit configuration with misconfiguration here.  if the admin forgets to set one of the paths then the integration will fail silently.

maybe a decent compromise would be to emphasize help text in the django admin next to these fields stating that integration will not work without specifying values.